test/plugins/orders.lua

```diff
@@ -1,26 +1,70 @@
 config.mode = 'fortress'
 
-TMP_FILE_NAME = 'tmp-test'
-TMP_FILE_PATH = ('dfhack-config/orders/%s.json'):format(TMP_FILE_NAME)
+local FILE_PATH_PATTERN = 'dfhack-config/orders/%s.json'
 
-function run_orders_import(file_content)
-    local f = io.open(TMP_FILE_PATH, 'w')
-    f:write(file_content)
-    f:close()
+local BACKUP_FILE_NAME = 'tmp-backup'
+local BACKUP_FILE_PATH = FILE_PATH_PATTERN:format(BACKUP_FILE_NAME)
 
+local TMP_FILE_NAME = 'tmp-test'
+local TMP_FILE_PATH = FILE_PATH_PATTERN:format(TMP_FILE_NAME)
+
+local function test_wrapper(test_fn)
+    -- backup and clear active orders
+    dfhack.run_command_silent{'orders', 'export', BACKUP_FILE_NAME}
+    dfhack.run_command_silent{'orders', 'clear'}
+    df.global.world.manager_order_next_id = 0
     return dfhack.with_finalize(
         function()
+            -- clear test orders, restore original orders, remove temp files
+            dfhack.run_command_silent{'orders', 'clear'}
+            df.global.world.manager_order_next_id = 0
+            dfhack.run_command_silent{'orders', 'import', BACKUP_FILE_NAME}
+            df.global.world.manager_order_next_id =
+                    #df.global.world.manager_orders
+            os.remove(BACKUP_FILE_PATH)
             os.remove(TMP_FILE_PATH)
         end,
-        function()
-            return dfhack.run_command_silent{'orders', 'import', TMP_FILE_NAME}
-        end
-    )
+        test_fn)
+end
+config.wrapper = test_wrapper
+
+-- returns export command result and exported file content
+function run_orders_export()
+    local _, result = dfhack.run_command_silent{'orders', 'export',
+                                                TMP_FILE_NAME}
+    local f = io.open(TMP_FILE_PATH, 'r')
+    return dfhack.with_finalize(
+        function() f:close() end,
+        function() return result, f:read('*all') end)
+end
+
+function run_orders_import(file_content)
+    local f = io.open(TMP_FILE_PATH, 'w')
+    f:write(file_content)
+    f:close()
+    return dfhack.run_command_silent{'orders', 'import', TMP_FILE_NAME}
+end
+
+local function normalize_whitespace(str)
+    return str:gsub('%s+', ' '):trim()
+end
+
+function check_export_success(expected_file_content)
+    local result, file_content = run_orders_export()
+    expect.eq(result, CR_OK)
+
+    -- ignore whitespace (otherwise the expected file content is impossible to
+    -- format properly in this file)
+    expect.eq(normalize_whitespace(expected_file_content),
+              normalize_whitespace(file_content))
 end
 
-function check_import_success(file_content)
+function check_import_success(file_content, comment, num_expected_orders)
```

Comment:
Minor thing: `comment` is unused. If it makes sense to include in failure messages, I suggest passing it through as the `comment` argument to the two `expect.eq` calls below to provide more context (or you could take the approach of `check_import_fail()` and create fancier error messages). Alternatively, the `comment` argument could be removed, given that test failure tracebacks _should_ be enough to isolate the cause of the failure now.

Reply:
Fixed. Passed `comment` through to the `expect.eq` calls, which was my original intention. The full tracebacks have definitely superseded comments for identifying which test failed, but I still find comments useful for reminding myself what the heck I'm actually testing if it's not completely obvious.